pkg/local_object_storage/writecache/storage.go

```diff
@@ -99,9 +100,14 @@ func (c *cache) deleteFromDB(keys [][]byte) error {
 	return c.db.Update(func(tx *bbolt.Tx) error {
 		b := tx.Bucket(defaultBucket)
 		for i := range keys {
+			has := b.Get(keys[i])
+			if has == nil {
+				return object.ErrNotFound
+			}
 			if err := b.Delete(keys[i]); err != nil {
 				return err
 			}
+			c.dbSize.Sub(uint64(len(has)))
```

Comment:
Same here, better to update it outside of tx.

Reply:
Done.